apps/wing/src/commands/test/test.ts

```diff
@@ -357,9 +364,102 @@ async function runTestsWithRetry(
   return results;
 }
 
+function inferSeverityOfEvent(trace: std.Trace): "error" | "warn" | "info" | "debug" | "verbose" {
+  if (trace.data.status === "failure") {
+    return "error";
+  }
+  if (trace.type === TraceType.LOG) {
+    return "info";
+  }
+  if (trace.type === TraceType.RESOURCE) {
+    return "debug";
+  }
+  if (trace.type === TraceType.SIMULATOR) {
+    return "verbose";
+  }
+  return "verbose";
+}
+
+const SEVERITY_STRING = {
+  error: "ERROR  ",
+  warn: "WARNING",
+  info: "INFO   ",
+  debug: "DEBUG  ",
+  verbose: "VERBOSE",
+};
+
+const LOG_STREAM_COLORS = {
+  error: chalk.red,
+  warn: chalk.yellow,
+  info: chalk.green,
+  debug: chalk.blue,
+  verbose: chalk.gray,
+};
+
 async function testSimulator(synthDir: string, options: TestOptions) {
   const s = new simulator.Simulator({ simfile: synthDir });
   const { clean, testFilter, retry } = options;
+
+  let outputStream: SpinnerStream | undefined;
+  if (options.stream) {
+    outputStream = new SpinnerStream(process.stdout, "Running tests...");
+
+    // As of this comment, each Wing test is associated with an isolated environment.
+    // (All resources for test #0 are in root/env0/..., etc.)
+    // This means we can use the environment number to map each environment # to a test name,
+    // so when we receive a trace from the simulator, we can infer which test it's associated with.
```

Comment:
Note: when we want to add tests that share the same resources, it might add some trickiness because it might not be as clear which test a log or event should be associated with.

I imagine if/when we have this kind of capability, we might be able to associate a name with the larger test suite:

```
describe "simple cluster" {
  let r = new redis.Redis();
  
  test "set" { ... }
  test "get" { ... }
}
```

And then all logs can be associated with "simple cluster", or if the tests are run sequentially, we can try and annotate them based on which test is currently logging. Anyway, just food for thought.